### ISSUE TITLE
Error when finishing training - AssertionError: Torch not compiled with CUDA enabled

### ISSUE THREAD

**Issue by eddjrn** [d8ahazard/sd_dreambooth_extension#10]
**Platform**: M1 Mac, RAM 32GB.
Generating images works with MPS instead of CUDA and with the Dreambooth extension training works with or without the CPU only option enabled.

**I tried to test training with 10 steps and it trains the model but when it finishes it throw the following message:**

```
Total target lifetime optimization steps = 10
 CPU: True Adam: False, Prec: no, Prior: True, Grad: True, TextTr: True
 Allocated: 0.0GB
 Reserved: 0.0GB

Steps: 100%|███████████████| 10/10 [04:32<00:00, 26.22s/it, loss=0.514, lr=5e-6]Error completing request
Arguments: ('pepe', '/Users/eduardoreyes/Desktop/backups/Weights/pepe/Processed', '/Users/eduardoreyes/Desktop/backups/Weights/Stable-Diffusion-Regularization-Images-person_ddim/person_ddim', 'photo of pepe person', 'photo of a person', '', '', 1.0, 7.5, 40.0, 440, 512, False, True, 1, 1, 1, 10, 1, True, 5e-06, False, 'constant', 0, False, 0.9, 0.999, 0.01, 1e-08, 1, 500, 500, 'no', True, '', True) {}
Traceback (most recent call last):
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/modules/ui.py", line 185, in f
    res = list(func(*args, **kwargs))
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/webui.py", line 54, in f
    res = func(*args, **kwargs)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/dreambooth.py", line 256, in start_training
    trained_steps = main(config)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 811, in main
    pipeline = pipeline.to("cuda")
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/diffusers/pipeline_utils.py", line 220, in to
    module.to(torch_device)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 987, in to
    return self._apply(convert)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 639, in _apply
    module._apply(fn)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 639, in _apply
    module._apply(fn)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 639, in _apply
    module._apply(fn)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 662, in _apply
    param_applied = fn(param)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/nn/modules/module.py", line 985, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
  File "/Users/eduardoreyes/Documents/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/cuda/__init__.py", line 221, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
```

**Comment by d8ahazard** [comment]
Can you try changing line 811 in /SD_DIRECTORY/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py from

`pipeline = pipeline.to("cuda")`

to

`pipeline = pipeline.to(accelerator.device)`

I believe that's the only line that needs changing, but the subsequent one where I autocast to cuda might also need to go.

**Comment by eddjrn** [comment]
With accelerator.device fixes the output error. Thank you

**Comment by d8ahazard** [comment]
Closed with f033e79.
